pep-9999.rst

```diff
@@ -0,0 +1,506 @@
+
+        aiohttp
+
+
+.. code-block::
+
+        aiohttp = {}
+
+**Simple Version Constraint**
+
+.. code-block::
+
+        aiohttp (>= 3.6.2, < 4.0.0)
+
+
+.. code-block::
+
+        aiohttp = { version = ">= 3.6.2, < 4.0.0" }
+
+
+.. note::
+
+        This can, for conciseness, be also represented as a string.
+
+        .. code-block::
+
+            aiohttp = ">= 3.6.2, < 4.0.0"
+
+
+
+Direct Reference Dependencies
+-----------------------------
+
+**URL Dependency**
+
+.. code-block::
+
+        aiohttp @ https://files.pythonhosted.org/packages/97/d1/1cc7a1f84097d7abdc6c09ee8d2260366f081f8e82da36ebb22a25cdda9f/aiohttp-3.6.2-cp35-cp35m-macosx_10_13_x86_64.whl
+
+
+.. code-block::
+
+        aiohttp = { url = "https://files.pythonhosted.org/packages/97/d1/1cc7a1f84097d7abdc6c09ee8d2260366f081f8e82da36ebb22a25cdda9f/aiohttp-3.6.2-cp35-cp35m-macosx_10_13_x86_64.whl" }
+
+**VCS Dependency**
+
+.. code-block::
+
+        aiohttp @ git+ssh://git@github.com/aio-libs/aiohttp.git@master
+
+
+.. code-block::
+
+        aiohttp = { git = "ssh://git@github.com/aio-libs/aiohttp.git", revision = "master" }
+
+
+Environment Markers
+-------------------
+
+.. code-block::
+
+        aiohttp (>= 3.6.1) ; python_version >= '3.8'
+
+
+.. code-block::
+
+        aiohttp = { version = ">= 3.6.1", markers = "python_version >= '3.8'" }
+
+
+A slightly extended example of the above, where a particular version of ``aiohttp`` is required based on the interpreter version.
+
+.. code-block::
+
+        aiohttp (>= 3.6.1) ; python_version >= '3.8'
+        aiohttp (>= 3.0.0, < 3.6.1) ; python_version < '3.8'
+
+
+.. code-block::
+
+        aiohttp = [
+            { version = ">= 3.6.1", markers = "python_version >= '3.8'" },
+            { version = ">= 3.0.0, < 3.6.1", markers = "python_version < '3.8'" }
+        ]
+
+
+Package Extras
+--------------
+
+**Specifying dependency for a package extra**
+
+.. code-block::
+
+        aiohttp (>= 3.6.2) ; extra == 'http'
+
+
+.. code-block::
+
+        aiohttp = { version = ">= 3.6.2", for-extra = "http" }
+
+**Using extras from a dependency**
+
+.. code-block::
+
+        aiohttp [speedups] (>= 3.6.2)
+
+
+.. code-block::
+
+        aiohttp = { version = ">= 3.6.2", extras = ["speedups"] }
+
+
+Complex Examples
+----------------
+
+**Version Constraint**
+
+.. code-block::
+
+        aiohttp [speedups] (>=3.6.2) ; python_version >= '3.8' and extra == 'http'
+
+
+.. code-block::
+
+        aiohttp = { version = ">= 3.6.2", extras = ["speedups"], markers = "python_version >= '3.8'", for-extra = "http" }
+
+
+**Direct Reference (VCS)**
+
+.. code-block::
+
+        aiohttp [speedups] @ git+ssh://git@github.com/aio-libs/aiohttp.git@master ; python_version >= '3.8' and extra == 'http'
+
+
+.. code-block::
+
+        aiohttp = { git = "ssh://git@github.com/aio-libs/aiohttp.git", revision = "master", extras = ["speedups"], markers = "python_version >= '3.8'", for-extra = "http" }
+
+
+Rejected Ideas
+==============
+
```

Comment:
Traditionally this section has subsections per rejected idea instead of a bulleted list, e.g.:
```restructuredtext
Using an array for `dependencies` instead of a table
-----------------------------------------------------
<exposition of what the idea was and why it was rejected>
```